### PR TITLE
implement task 5

### PR DESCRIPTION
**Pull request by Julia-ivv**
BE importProductsFile - https://a7iew7bnge.execute-api.eu-west-1.amazonaws.com/dev/import?name=products-for-import.csv
FE - https://d2e65jaievfvyp.cloudfront.net/

## Main tasks
- **done** - File **serverless.yml** contains configuration for **importProductsFile** function
- **done** - The **importProductsFile** lambda function returns a correct response which can be used to upload a file into the **S3** bucket
- **done** - Frontend application is integrated with **importProductsFile** lambda
- **done** - The **importFileParser** lambda function is implemented and **serverless.yml** contains configuration for the lambda

## Additional (optional) tasks
- **+1** - **async/await** is used in lambda functions - done
- **+1** - At the end of the **stream** the lambda function should move the file from the **uploaded** folder into the **parsed** folder - done

### CODE REVIEW

File: import-service/handler.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+const S3 = require('aws-sdk/clients/s3');
+const csv = require('csv-parser');
+
+const headers = {
+    'Content-Type': 'application/json',
+    'Access-Control-Allow-Origin': '*',
+    'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept',
+    'Access-Control-Allow-Methods': 'POST, GET, OPTIONS, PATCH, PUT',
```
Comment (kotmatpockuh) [inline import-service/handler.js:8]: Doubt you need all these methods :)

File: import-service/handler.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            const s3 = new S3({ region: 'eu-west-1', signatureVersion: 'v4' });
+            const fileName = event.queryStringParameters.name;
+            const params = {
+                Bucket: BUCKET,
+                Key: `uploaded/${fileName}`,
+                Expires: 60,
+                ContentType: 'text/csv'
+            };
+        
+            return await new Promise((resolve, reject) => {
```
Comment (kotmatpockuh) [inline import-service/handler.js:24]: weird async/await implementation :)

I would expect something like this: ```return await s3.getSignedUrlPromise...```

File: import-service/handler.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                });
+            });
+        } catch(error) {
+            return {
+                statusCode: 500,
+                body: JSON.stringify({message: 'Internal Server Error'}),
+            }
+        };
+    },
+    importFileParser: (event) => {
```
Comment (kotmatpockuh) [inline import-service/handler.js:44]: normally you should split this into separated files ;)

File: import-service/handler.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        try {
+            const s3 = new S3({ region: 'eu-west-1', signatureVersion: 'v4' });
+            event.Records.forEach(record => {
+                const s3Stream = s3.getObject({
+                    Bucket: BUCKET,
+                    Key: record.s3.object.key,
+                }).createReadStream();
+
+                s3Stream.pipe(csv())
```
Comment (kotmatpockuh) [inline import-service/handler.js:53]: you can add ```await s3Stream.pipe(csv())...```, move _copyObject & deleteObject outside_ and remove nested await-s 🤔

File: import-service/handler.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                        }).promise();
+                        await s3.deleteObject({
+                            Bucket: BUCKET,
+                            Key: record.s3.object.key
+                        }).promise();
+                    } catch(error) {
+                        console.log('error', error);
+                    }
+                });
+            });
```
Comment (kotmatpockuh) [inline import-service/handler.js:71]: I guess return is missing?

**Comment by kotmatpockuh** [comment]
7/8
am I correct?

**Comment by Julia-ivv** [comment]
> 7/8
> am I correct?

yes, that's right, thanks for the comments

**Comment by numatay** [comment]
Hi @Julia-ivv Do you have a link to your S3 `uploaded & parsed` folders so I could check that the files were moved accordingly?
The name of your bucket will suffice I guess.

**Comment by Julia-ivv** [comment]
> Hi @Julia-ivv Do you have a link to your S3 `uploaded & parsed` folders so I could check that the files were moved accordingly?
> The name of your bucket will suffice I guess.

The bucket name is s3://julia-ivv-stall-import,

**Comment by numatay** [comment]
Very concise implementation. It works as expected, however, I would take a look at the comments mentioned above.
7 pts.

**Pull request merged**